lib-ui-compose/src/androidMain/kotlin/com/ustadmobile/libuicompose/components/UstadContactPickButton.kt

```diff
@@ -0,0 +1,158 @@
+package com.ustadmobile.libuicompose.components
+
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.provider.ContactsContract
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.ustadmobile.core.MR
+import com.ustadmobile.libuicompose.util.ext.defaultItemPadding
+import dev.icerock.moko.resources.compose.stringResource
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+
+@Composable
+actual fun UstadContactPickButton(
+    onContactPicked: (String?) -> Unit
+
+) {
+    val context = LocalContext.current
+    var dialogVisible: Boolean by remember {
+        mutableStateOf(false)
+    }
+
+    /**
+     * to get  contact number from contact Address
+     */
+    val pickContactLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.StartActivityForResult()
+    ) { result ->
+        if (result.resultCode == Activity.RESULT_OK) {
+            val data: Intent? = result.data
+            val contactUri: Uri? = data?.data
+            if (contactUri != null) {
+                val projection = arrayOf(ContactsContract.CommonDataKinds.Phone.NUMBER)
+                val cursor = context.contentResolver.query(contactUri, projection, null, null, null)
+                CoroutineScope(Dispatchers.IO).launch {
```

Comment:
Instead of creating a new coroutine scope - use the coroutine scope that is provided from compose so that any task would be cancelled automatically if the user navigates away.

eg.

val scope = rememberCoroutineScope()

then to run something off the main thread
```
scope.launch {
   withContext(Dispatchers.IO) {
       //do something - will be cancelled automatically if user went away
   } 
}
```